dm/master/shardddl/optimist_test.go

```diff
@@ -254,8 +254,11 @@ func (t *testOptimist) testOptimist(c *C, restart int) {
 	// wait operation for i11 become available.
 	opCh := make(chan optimism.Operation, 10)
 	errCh := make(chan error, 10)
-	ctx2, cancel2 := context.WithTimeout(ctx, watchTimeout)
-	optimism.WatchOperationPut(ctx2, etcdTestCli, i11.Task, i11.Source, i11.UpSchema, i11.UpTable, rev1, opCh, errCh)
+	ctx2, cancel2 := context.WithCancel(ctx)
+	go optimism.WatchOperationPut(ctx2, etcdTestCli, i11.Task, i11.Source, i11.UpSchema, i11.UpTable, rev1, opCh, errCh)
+	utils.WaitSomething(10, watchTimeout, func() bool {
```

Comment:
should use assert?

Reply:
assert is in lines below https://github.com/pingcap/dm/pull/785/files#diff-2bb523adf53949e58e838fc0685db092R265 , github doesn't display it.

these code just dynamic increase waiting time, for best case, it waits `watchTimeout ` which is same as original code, else it trys to wait more. after timeout, assertion is called.